src/index.tsx

```diff
@@ -28,165 +44,132 @@ interface AudioPosition {
+            if (playerRef.current) {
+                // don't do anything if we're asked to reload the same source
+                // @ts-ignore the _src argument actually exists
+                if (playerRef.current._src === src) return
+                wasPlaying = playerRef.current.playing()
+                // destroys the previous player
+                playerRef.current.unload()
             }
 
-            setLoading(true)
-            player.current = new Howl({
+            // create a new player
+            const howl = new Howl({
                 src,
-                onload() {
-                    setLoading(false)
-                    setPlaybackReady(true)
+                format,
+                autoplay: wasPlaying || autoplay, // continues playing next song
+                onload: () => void setLoading(false),
+                onplay: () => {
+                    // prevents howl from playing the same song twice
+                    if (!howl.playing()) return
+                    setPlaying(true)
+                    setStopped(false)
+                },
+                onpause: () => void setPlaying(false),
+                onstop: () => {
+                    setStopped(true)
+                    setPlaying(false)
+                },
+                onplayerror: (_id, error) => {
+                    setError(new Error("[Play error] " + error))
+                    setPlaying(false)
+                    setStopped(true)
                 },
-                onloaderror(id, error) {
+                onloaderror: (_id, error) => {
+                    setError(new Error("[Load error] " + error))
                     setLoading(false)
-                    setPlaybackReady(false)
-                    setError(new Error(error))
-                }
+                },
             })
-        },
-        [player, playbackReady]
-    )
-
-    const play = React.useCallback(() => {
-        if (!player.current.playing()) {
-            player.current.play()
-        }
-    }, [player])
 
-    const pause = React.useCallback(() => {
-        player.current.pause()
-    }, [player])
-
-    const stop = React.useCallback(() => {
-        player.current.stop()
-    }, [player])
-
-    const seek = React.useCallback(
-        (pos: number) => {
-            if (playbackReady) {
-                player.current.seek(pos)
-            }
+            setPlayer(howl)
+            playerRef.current = howl
         },
-        [player, playbackReady]
+        []
     )
 
-    const isPlaying = React.useCallback(() => {
-        if (!playbackReady) return false
-        return player.current.playing()
-    }, [player, playbackReady])
-
-    const mute = React.useCallback(
-        (muted: boolean) => {
-            if (playbackReady) {
-                player.current.mute(muted)
-            }
-        },
-        [player, playbackReady]
-    )
+    useEffect(() => {
+        // unload the player on unmount
+        return () => {
```

Comment:
rather than being called on unmount this function is invoked at the start of the next rerender which seems like it could interrupt the audio if the AudioPlayerProvider is rerendered

Reply:
Hmm unless I'm missing something the dependency here is set to `[]` which only happens on unmount.